Building a Smart Agent/smart_agent.py

```diff
@@ -71,7 +71,7 @@ def choose_action(self, observation):
             # some actions have the same value
             state_action = state_action.reindex(np.random.permutation(state_action.index))
             
-            action = state_action.argmax()
+            action = state_action.values.argmax()
```

Comment:
This change stops my code from properly selecting an action, what is this change for?

Reply:
I was having trouble running your example code -- I haven't spent enough time pinpointing this issue to tell you exactly what's wrong, but I did find a quick fix [here](https://github.com/pandas-dev/pandas/issues/18021).

From what I skimmed it has to do with `argmax` being deprecated in a newer version of pandas' `dtype`. The specific error I was getting is `TypeError: reduction operation 'argmax' not allowed for this dtype`.

I can investigate this further if you think it's worth pursuing, we can simply close the PR as a non-issue, or perhaps the best solution is to include a `requirements.txt` to lock in the versions of things.

Reply:
Hi @clrobert it turns out you can use `action = state_action.idxmax()`

Reply:
Awesome, thanks 👍